sharedbuffers/mapped_struct.py

```diff
@@ -1708,6 +1712,65 @@ def _c_search_hkey_ui64(hkey, pindex, stride0, length, hint):
 
     @cython.cfunc
     @cython.locals(
+        hkey = cython.longlong,
+        mkey = cython.longlong,
+        lo = cython.size_t, hi = cython.size_t, length = cython.size_t,
+        mid = cython.size_t, mid2 = cython.size_t, stride0 = cython.size_t, hint = cython.size_t,
+        pindex = cython.p_char, skip = cython.size_t)
+    @cython.returns(cython.size_t)
+    def _c_search_hkey_i64(hkey, pindex, stride0, length, hint):
+        hi = length
+        lo = 0
+        if lo < hi:
+            # First iteration a quick guess assuming uniform distribution of keys
+            mid = min(hint, hi-1)
```

Comment:
hint is an externally supplied index. I do need it to avoid a segmentation fault.